FilterQueryLanguage.Core/Models/QueryTree.cs

```diff
@@ -0,0 +1,109 @@
+namespace FilterQueryLanguage.Core.Models
+{
+     public class QueryTree
+    {
+        private QueryNode _root;
+        public QueryTree()
+        {
+            _root = null;
+        }
+
+        public void Insert(FilterQueryLogicalOperator node)
+        {
+            if (_root == null)
+            {
+                _root = new QueryNode(node);
+                return;
+            }
+            //otherwise recurse down
+            InsertRec(_root, new QueryNode(node));
+        }
+
+        public void Insert(PredicateModel node)
+        {
+            if (_root == null)
+            {
+                _root = new QueryNode(node);
+                return;
+            }
+            //otherwise recurse down
+            InsertRec(_root, new QueryNode(node));
+        }
+
+        public void Insert(QueryNode node)
+        {
+            if (_root == null)
+            {
+                _root = node;
+                return;
+            }
+            //otherwise recurse down
+            InsertRec(_root, node);
+        }
+
+        private void InsertRec(QueryNode root, QueryNode newNode)
+        {
+            if (root == null)
+            {
+                root = newNode;
+            }
+
+            if (root.LogicalOperatorNode && newNode.LogicalOperatorNode && root.Left != null && root.Right != null)
+            {
+                _root = newNode;
+                _root.Left = root;
+            }
+
+            if (root.LogicalOperatorNode && newNode.LogicalOperatorNode && root.Left != null && root.Right == null)
+            {
+                _root.Right = newNode;
+            }
+
+            if (root.PredicateNode && newNode.LogicalOperatorNode)
+            {
+                InsertRec(newNode, root);
+            }
+
+
```

Comment:
I forget if we're suppose to comment on formatting or not, but there's 2 spaces here